pkg/runtime/python/python.go

```diff
@@ -345,150 +162,328 @@ func (r *PythonRuntime) Run(ctx context.Context, input *runtime.RunInput) (runti
+		// Extract the tar.gz file
+		cmd := process.CommandContext(ctx, "tar", "-xzf", file, "-C", input.Out())
+		cmd.Dir = input.Out()
+		err = cmd.Run()
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract tar.gz file: %v", err)
 		}
 
-		// Move up one directory
-		parentDir := filepath.Dir(dir)
-		if parentDir == dir {
-			// Reached the root directory
-			break
+		// Get the directory name without version number
+		dirName := strings.TrimSuffix(filepath.Base(file), ".tar.gz")
+		lastHyphen := strings.LastIndex(dirName, "-")
+		baseName := dirName[:lastHyphen]
+
+		extractedDir := filepath.Join(input.Out(), dirName)
+		targetDir := filepath.Join(input.Out(), baseName)
+
+		// Check if the package has a src/{package_name} structure
+		srcPath := filepath.Join(extractedDir, "src", baseName)
+		if _, err := os.Stat(srcPath); err == nil {
+			// Remove old directory if it exists
+			if err := os.RemoveAll(targetDir); err != nil {
+				return nil, fmt.Errorf("failed to remove old directory: %v", err)
+			}
+			// Move the contents from src/{package_name} directly to the target
+			if err := os.Rename(srcPath, targetDir); err != nil {
+				return nil, fmt.Errorf("failed to move src directory contents: %v", err)
+			}
+			// Clean up the original extracted directory
+			if err := os.RemoveAll(extractedDir); err != nil {
+				return nil, fmt.Errorf("failed to clean up extracted directory: %v", err)
+			}
+		} else {
+			// Handle the regular case (no src directory)
+			if err := os.RemoveAll(targetDir); err != nil {
+				return nil, fmt.Errorf("failed to remove old directory: %v", err)
+			}
+			if err := os.Rename(extractedDir, targetDir); err != nil {
+				return nil, fmt.Errorf("failed to rename directory: %v", err)
+			}
 		}
-		dir = parentDir
 	}
-	return "", fmt.Errorf("pyproject.toml not found")
-}
 
-func getPythonFiles(filePath string) ([]string, error) {
-	// Get the absolute path of the file
-	absPath, err := filepath.Abs(filePath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get absolute path: %v", err)
+	// 4. Remove the tar.gz files (non-recursive)
+	for _, file := range files {
+		err = os.Remove(file)
+		if err != nil {
+			return nil, fmt.Errorf("failed to remove tar.gz file: %v", err)
+		}
 	}
 
-	// Get the directory of the file
-	dir := filepath.Dir(absPath)
+	// If making a zip build or a local build then we need to install the dependencies and adjust the handler path
+	if !input.IsContainer || input.Dev {
 
-	var pythonFiles []string
-
-	// Walk through the directory
-	err = filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		// 5. Install the dependencies as a target
+		args := []string{"pip", "install", "-r", outputRequirementsFile, "--target", input.Out()}
+		if !input.Dev {
+			// If we are not in dev mode then we need to install the dependencies for the target platform
+			// which is amazon linux
+			args = append(args, "--python-platform", "linux")
```

Comment:
Thanks for the change but it should also depends on the cpu: 
- x86_64-unknown-linux-gnu when deploying to x86_64
- aarch64-unknown-linux-gnu when deploying to arm64